### DISCUSSION TITLE
Best way to export volumetric loss at all HFSS grid points?

### DISCUSSION THREAD

**Discussion by khoopes** [ansys/pyaedt#4115] [Q&A] [answered]
I am trying to export volumetric (and surface) loss for my HFSS model for use in another code. What I would really like is to export a csv of x,y,z,loss at everywhere the solver has information for this. Here is what I have tried so far (both via pyaedt and manually in aedt):

Create a plot of volumetric loss density and export as an AEDTPLT file. This works and I was able to parse the aedtplt file to get element and node locations. For my model it has 10 nodes per element and gives the x,y,z location of each node. But it gives 20 loss values per element. In the HFSS documentation it says to reference http://www.vki.com/2013/Support/docs/vistools-1.html#1.2. which I found here: https://web.archive.org/web/20230129125615/http://www.vki.com/2013/Support/docs/vistools-1.html but that also has 10 nodes per element. Where do the other 10 values come in?

Use field calculator to "Write" volumetric loss. This writes 20 values per element (just like the aedtplt file) but no information on where these values are geometrically, if this just also contained the x,y,z location for each loss value, this would be perfect.

Use field calculator to "Export" volumetric loss. This is a non starter as it requires a gird, and I want to use the grid that HFSS already has.

Any ideas on how to do this?

Thanks,

Kevin

**Comment by khoopes** [comment]
More notes on this, it seems the solution order has something to do with what is going on:

I wrote a new AEDTPLT file for each solution order:
Zero Order - 10 nodes per element, 10 solution values per element
First Order - 10 nodes per element, 10 solution values per element
Second Order - 10 nodes per element, 20 solution values per element

This is confusing to me. shouldn't the Zero order solution have only 4 values per element, just at the vertices?

**Reply by Samuelopez-ansys** [reply, marked as answer]
Hi @khoopes,

This forum is for PyAEDT questions, please submit a ticket in the Ansys customer portal if you need Ansys product support.
